android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateUtils.java

```diff
@@ -3,11 +3,66 @@
 import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.WritableMap;
 
+import org.json.JSONArray;
+
+import java.io.ByteArrayInputStream;
 import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.io.InputStream;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collections;
 
 public class CodePushUpdateUtils {
 
+    private static void addContentsOfFolderToManifest(String folderPath, String pathPrefix, ArrayList<String> manifest) {
+        File folder = new File(folderPath);
+        File[] folderFiles = folder.listFiles();
+        for (File file : folderFiles) {
+            String fullFilePath = file.getAbsolutePath();
+            String relativePath = (pathPrefix.isEmpty() ? "" : (pathPrefix + "/")) + file.getName();
+            if (file.isDirectory()) {
+                addContentsOfFolderToManifest(fullFilePath, relativePath, manifest);
+            } else {
+                try {
+                    // Substring 1 because appendPathComponent creates a "/" prefix to the relativePath.
+                    manifest.add(relativePath + ":" + computeHash(new FileInputStream(file)));
+                } catch (FileNotFoundException e) {
+                    // Should not happen.
+                    throw new CodePushUnknownException("Unable to compute hash of update contents.", e);
+                }
+            }
+        }
+    }
+
+    private static String computeHash(InputStream dataStream) {
+        MessageDigest messageDigest = null;
+        DigestInputStream digestInputStream = null;
+        try {
+            messageDigest = MessageDigest.getInstance("SHA-256");
+            digestInputStream = new DigestInputStream(dataStream, messageDigest);
+            byte[] byteBuffer = new byte[1024 * 8];
+            while (digestInputStream.read(byteBuffer) != -1);
+        } catch (NoSuchAlgorithmException | IOException e) {
+            // Should not happen.
+            throw new CodePushUnknownException("Unable to compute hash of update contents.", e);
+        } finally {
+            try {
+                if (digestInputStream != null) digestInputStream.close();
+                if (dataStream != null) dataStream.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        byte[] hash = messageDigest.digest();
+        return String.format("%064x", new java.math.BigInteger(1, hash));
```

Comment:
Could we use the "android.util.Base64" class here instead of doing this? This article seems to use that solution: http://karanbalkar.com/2013/05/tutorial-28-implement-sha1-and-md5-hashing-in-android/